src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapSessionOptions.Linux.cs

```diff
@@ -71,5 +106,14 @@ private void SetBoolValueHelper(LdapOption option, bool value)
 
             ErrorChecking.CheckAndSetLdapError(error);
         }
+
+        private void SetStringOptionHelper(LdapOption option, string value)
+        {
+            if (_connection._disposed) throw new ObjectDisposedException(GetType().Name);
```

Comment:
nit: why use reflection here?  Couldn't we just do a `nameof(LdapSessionOptions)`.  It's an existing pattern so it's not new to this PR but it stuck out as unusual use of `GetType()`

Reply:
This is following the pattern done throughout the LDAP code (see above in `SetBoolValueHelper`) so that the proper derived class name is shown in the exception.

However, this is done even in cases where it may not be possible to have a derived class. We could clean this up, but IMO that should be in another PR.